.github/actions/firebase-trigger/index.js

```diff
@@ -0,0 +1,41 @@
+const core = require('@actions/core');
+const github = require('@actions/github');
+const admin = require('firebase-admin');
+const firebase = require('firebase');
+// This functions initializes the firebase.
+function initFirebase() {
+	core.info("Initialized Firebase Admin Connection");
+	const credentials = core.getInput('credentials'); 
+	admin.initializeApp({
+		credential: admin.credential.cert(JSON.parse(credentials)),
+		databaseURL: core.getInput('databaseUrl'),
+	});
+}
+// This function gets the output of the other action to get the github files URL.
+function getValue() {
+	core.info('Getting the value');
+	const value = core.getInput('value');
+	return value;
+}
+// This function updates the firestore 
+function updateFirestoreDatabase(collection, value) {
+	const db = admin.firestore();
+	const document = core.getInput('doc');
+	// TODO : to check the status of the change and call the function accordingly.
+	for (let change of value) {
+		var pathToValue = change.path;
+		var valuetoUpdate = change.value;
+		let temp = {};
+		temp[pathToValue] = valuetoUpdate;
+		db.collection(collection).doc(document).update(temp);
+	}
+}
+// main function of the script calls all the functions.
+function processJavascriptAction() {
+	initFirebase();
+	const collection = core.getInput('collection');
+	const value = getValue();
+	var jsonValue = JSON.parse(value);
+	updateFirestoreDatabase(collection, jsonValue);
+}
+processJavascriptAction();
```

Comment:
```suggestion
function run() {
	initFirebase();
	const collection = core.getInput('collection');
	const value = getValue();
	var jsonValue = JSON.parse(value);
	updateFirestoreDatabase(collection, jsonValue);
}

run();
```
We should change the name of the function to make it consistent with my action